lib/synapse/config_generator/haproxy.rb

```diff
@@ -814,7 +814,7 @@ def initialize(opts)
       @opts['do_writes'] = true unless @opts.key?('do_writes')
       @opts['do_socket'] = true unless @opts.key?('do_socket')
       @opts['do_reloads'] = true unless @opts.key?('do_reloads')
-
+      @opts['use_nerve_weights'] = true if @opts.key?('use_nerve_weights')
```

Comment:
what is the expected default behavior when 'use_nerve_weights' is not set in the config? should that be false?

Reply:
If it is not set in the config, the idea is it should be false.  This is to provide backwards compatibility for this change.  Default to the old behavior.

Reply:
ok. so the logic here is to leave it as nil if the key is not set. to make it false by default, shall we do
@opts['use_nerve_weights'] = false unless @opts.key?('use_nerve_weights')

Reply:
I went ahead and removed this.  This was an artifact of the original change that was not backwards compatible and i swapped this if statement to make the change backwards compatible.  And in the current state, it effectively does nothing, if anything it adds confusion to how the `use_nerve_weights` flag is used.